File: mobile/hika_app/lib/features/profile/data/user_profile.dart
class UserProfile {
  const UserProfile({
    required this.userId,
    required this.email,
    required this.emailVerified,
    required this.firstName,
    required this.lastName,
    required this.phoneVerified,
    required this.memberSinceUtc,
    required this.completedTripCount,
    this.photoUrl,
    this.phoneNumber,
    this.averageRating,
  });

  factory UserProfile.fromJson(Map<String, dynamic> json) => UserProfile(
    userId: json['userId'] as String,
    email: json['email'] as String,
    emailVerified: json['emailVerified'] as bool,
    firstName: json['firstName'] as String,
    lastName: json['lastName'] as String,
    photoUrl: json['photoUrl'] as String?,
    phoneNumber: json['phoneNumber'] as String?,
    phoneVerified: json['phoneVerified'] as bool,
    memberSinceUtc: DateTime.parse(json['memberSinceUtc'] as String),
    averageRating: (json['averageRating'] as num?)?.toDouble(),
    completedTripCount: json['completedTripCount'] as int,
  );

  final String userId;
  final String email;
  final bool emailVerified;
  final String firstName;
  final String lastName;
  final String? photoUrl;
  final String? phoneNumber;
  final bool phoneVerified;
  final DateTime memberSinceUtc;
  final double? averageRating;
  final int completedTripCount;

  String get fullName => '$firstName $lastName';

  String get initials {
    final f = firstName.isNotEmpty ? firstName[0] : '';
    final l = lastName.isNotEmpty ? lastName[0] : '';
    return '$f$l'.toUpperCase();
  }
}

File: backend/src/Hika.Domain/Users/UserProfile.cs
using Hika.Domain.Common;

namespace Hika.Domain.Users;

/// <summary>
/// The profile/domain data for a user, separate from <c>ApplicationUser</c> (which is an
/// Infrastructure/Identity persistence concern — see docs/architecture.md §4). Keyed by the
/// same Guid as the identity user (a shared-primary-key 1:1 relationship), referenced only
/// by that Guid, never by a navigation property, so this project has no dependency on Identity.
/// </summary>
public sealed class UserProfile : Entity
{
    public string FirstName { get; private set; }

    public string LastName { get; private set; }

    public string? PhotoUrl { get; private set; }

    public string? PhoneNumber { get; private set; }

    public DateTimeOffset? PhoneVerifiedAtUtc { get; private set; }

    public DateTimeOffset? EmailVerifiedAtUtc { get; private set; }

    public DateTimeOffset MemberSinceUtc { get; private set; }

    public decimal? AverageRating { get; private set; }

    public int CompletedTripCount { get; private set; }

    private UserProfile(Guid userId, string firstName, string lastName)
    {
        Id = userId;
        FirstName = firstName;
        LastName = lastName;
        MemberSinceUtc = DateTimeOffset.UtcNow;
        CompletedTripCount = 0;
    }

    /// <remarks>Required by EF Core materialization.</remarks>
    private UserProfile()
    {
        FirstName = string.Empty;
        LastName = string.Empty;
    }

    public static UserProfile Create(Guid userId, string firstName, string lastName) =>
        new(userId, firstName, lastName);

    public void UpdateName(string firstName, string lastName)
    {
        FirstName = firstName;
        LastName = lastName;
    }

    public void UpdatePhoto(string? photoUrl) => PhotoUrl = photoUrl;

    /// <summary>Changing the phone number always requires re-verification.</summary>
    public void SetPhoneNumber(string phoneNumber)
    {
        if (phoneNumber == PhoneNumber)
        {
            return;
        }

        PhoneNumber = phoneNumber;
        PhoneVerifiedAtUtc = null;
    }

    public void MarkPhoneVerified() => PhoneVerifiedAtUtc = DateTimeOffset.UtcNow;

    public void MarkEmailVerified() => EmailVerifiedAtUtc = DateTimeOffset.UtcNow;

    public void RecordCompletedTripReview(decimal newRating)
    {
        var totalRatingPoints = (AverageRating ?? 0m) * CompletedTripCount;
        CompletedTripCount++;
        AverageRating = decimal.Round((totalRatingPoints + newRating) / CompletedTripCount, 2);
    }
}
